### PR TITLE
Adds RunLLM chat widget to MLflow's doc site

### PR DESCRIPTION
**Pull request by likawind**
<details><summary>&#x1F6E0 DevTools &#x1F6E0</summary>
<p>

#### Install mlflow from this PR

```
pip install git+https://github.com/mlflow/mlflow.git@refs/pull/13122/merge
```

#### Checkout with GitHub CLI

```
gh pr checkout 13122
```

</p>
</details>

### Related Issues/PRs
N/A

### What changes are proposed in this pull request?

This PR adds RunLLM chat widget to MLFlow's doc page

### How is this PR tested?

- [ ] Existing unit/integration tests
- [ ] New unit/integration tests
- [x] Manual tests (WIP testing in docs preview environment)

### Does this PR require documentation update?

- [x] No. You can skip the rest of this section.
- [ ] Yes. I've updated:
  - [ ] Examples
  - [ ] API references
  - [ ] Instructions

### Release Notes
We added RunLLM chat widget for MLFlow users to easily explore the docs through a chat interface.
#### Is this a user-facing change?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

#### What component(s), interfaces, languages, and integrations does this PR affect?

Components

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/deployments`: MLflow Deployments client APIs, server, and third-party Deployments integrations
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>

#### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [x] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

#### Should this PR be included in the next patch release?

`Yes` should be selected for bug fixes, documentation updates, and other small changes. `No` should be selected for new features and larger changes. If you're unsure about the release classification of this PR, leave this unchecked to let the maintainers decide.

<details>
<summary>What is a minor/patch release?</summary>

- Minor release: a release that increments the second part of the version number (e.g., 1.2.0 -> 1.3.0).
  Bug fixes, doc updates and new features usually go into minor releases.
- Patch release: a release that increments the third part of the version number (e.g., 1.2.0 -> 1.2.1).
  Bug fixes and doc updates usually go into patch releases.

</details>

- [x] Yes (this PR will be cherry-picked and included in the next patch release)
- [ ] No (this PR will be included in the next minor release)

### CODE REVIEW

**Review by daniellok-db — APPROVED** [review]
LGTM!

**Comment by harupy** [comment]
The reason it doesn't work in the preview is it only works on mlflow.org?

image

**Comment by daniellok-db** [comment]
cc @likawind to confirm if the widget is only expected to work on mlflow.org

**Comment by cw75** [comment]
Hi @daniellok-db @harupy, I just added the test site's domain to our allowed list and verified that it works fine (you may need to clear the browser cache when testing).

Screenshot 2024-09-11 at 1 10 50 AM

**Review by harupy — APPROVED** [review]
LGTM once https://github.com/mlflow/mlflow/pull/13123/files#r1753914058 is resolved!

**Comment by cw75** [comment]
Hi @harupy, we have made the requested changes. Thanks!

File: docs/source/_static/runllm.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    script.id = "runllm-widget-script"
+
+    script.src = "https://widget.runllm.com";
+
+    script.setAttribute("runllm-keyboard-shortcut", "Mod+j"); // cmd-j or ctrl-j to open the widget.
+    script.setAttribute("runllm-name", "MLflow");
+    script.setAttribute("runllm-position", "BOTTOM_RIGHT");
+    script.setAttribute("runllm-assistant-id", "116");
+    script.setAttribute("runllm-theme-color", "#008ED9");
+    script.setAttribute("runllm-brand-logo", "_static/favicon.ico");
```
Comment (harupy) [inline docs/source/_static/runllm.js:13]:
```suggestion
    script.setAttribute("runllm-brand-logo", "https://mlflow.org/img/mlflow-favicon.ico");
```

**Review by harupy — APPROVED** [review]
LGTM

**Pull request merged**